### PR TITLE
[OSCD] Create powershell_cmdline_reversed_strings

### PR DESCRIPTION
**Pull request by vburov**

### CODE REVIEW

File: rules/windows/powershell/powershell_cmdline_reversed_strings.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+description: Detects the PowerShell command lines with reversed strings
+status: experimental
+references:
+    - https://2019.offzone.moscow/ru/report/hunting-for-powershell-abuses/
+    - https://speakerdeck.com/heirhabarov/hunting-for-powershell-abuse?slide=66
+tags:
+    - attack.defense_evasion
+    - attack.t1027
+author: Teymur Kheirkhabarov (idea), Vasiliy Burov (rule), oscd.community
+date: 2020/11/10
```
Comment (e6e6e) [inline rules/windows/powershell/powershell_cmdline_reversed_strings.yml:12]: Please change the date format (correct format - YYYY/MM/DD).

File: rules/windows/powershell/powershell_cmdline_reversed_strings.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+logsource:
+    category: process_creation
+    product: windows
+detection:
+    selection:
+        CommandLine|contains:
```
Comment (e6e6e) [inline rules/windows/powershell/powershell_cmdline_reversed_strings.yml:18]: I think it is appropriate to add condition (to reduce the amount falsepositives):
```
filter:
	Image|endswith: '\powershell.exe'
```
What do you feel about it?

File: rules/windows/powershell/powershell_cmdline_reversed_strings.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
+title: Suspicious PowerShell Cmdline
+id: b6b49cd1-34d6-4ead-b1bf-176e9edba9a4
+description: Detects the PowerShell command lines with reversed strings
+status: experimental
+references:
+    - https://2019.offzone.moscow/ru/report/hunting-for-powershell-abuses/
+    - https://speakerdeck.com/heirhabarov/hunting-for-powershell-abuse?slide=66
+tags:
+    - attack.defense_evasion
+    - attack.t1027
```
Comment (e6e6e) [inline rules/windows/powershell/powershell_cmdline_reversed_strings.yml:10]: Hello Vasiliy!
Thank you for your contribution.
Please add this tags:
 - attack.execution
 - attack.t1059.001

Comment (vburov) [reply rules/windows/powershell/powershell_cmdline_reversed_strings.yml:10]: Hi! Done.

File: rules/windows/powershell/powershell_cmdline_reversed_strings.yml (hunk 1 above)
Comment (vburov) [reply rules/windows/powershell/powershell_cmdline_reversed_strings.yml]: Done.

File: rules/windows/powershell/powershell_cmdline_reversed_strings.yml (hunk 2 above)
Comment (vburov) [reply rules/windows/powershell/powershell_cmdline_reversed_strings.yml:20]: Hi! Thanks for suggestion! I think that in some cases the "cmd.exe" may be as parent process.

Comment (e6e6e) [reply rules/windows/powershell/powershell_cmdline_reversed_strings.yml:20]: It seems to me that the templates are not very specific for PowerShell. For example, they are relevant for [VBScript](https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualbasic.strings.strreverse?view=netcore-3.1).
Of course such a case is possible ([for example](https://www.ired.team/offensive-security/code-execution/powershell-without-powershell)).
I think that in most cases _powershell_ will be in the _Image_. (I cheked it)

So, I have two suggestions:
1. Do not mark in the title of the rule that it only detects _powershell_.
2. Leave the current rule name, but add _powershell.exe_ to the filter.

In any case, the rule looks good now. =)
@yugoslavskiy

Comment (vburov) [reply rules/windows/powershell/powershell_cmdline_reversed_strings.yml]: Ok, I will not try to cover all possible cases :-) I added the powershell.exe as value for Image field.

Comment (e6e6e) [reply rules/windows/powershell/powershell_cmdline_reversed_strings.yml]: We did a good job! =)
I think it's ready to be merged. @yugoslavskiy

**Comment by yugoslavskiy** [comment]
Thank you very much, @vburov! Well done!

**Review by yugoslavskiy — APPROVED** [review]
@thomaspatzke @Neo23x0 guys, I believe that this PR could be merged (:
please let us know what do you think about it if there is anything you would like to change.

**Pull request merged**